### PR TITLE
Adding release notes for 0.9.8 (hotfix).

### PR DESCRIPTION
**Pull request by jiajunwang**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:

NA, doc change only

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:

This PR adds 0.9.8 release notes and other documents.
Release 0.9.8 is a hotfix version based on 0.9.7.

Note that we are not adding 0.9.8 into the main page before vote completes. There might be some updates later based on the discussions.

Please refer to the following snapshot for the release detail.

Screenshot from 2020-10-13 16-20-33

### Tests

- [X] The following tests are written for this issue:

NA

- [X] The following is the result of the "mvn test" command on the appropriate module:

mvn site:site
mvn site:stage

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

File: website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  In this release, Helix provides several interface improvements and bug fixes:
+
+[]
+
+* Key Note for Helix Release
+
+** Critical Bug Fixes and Craftsmanship Improvements
+
+    In this release, we fixed one critical bugs in Helix ZkClient. There are also several fixes for craftsmanship and interfaces improvements.
```
Comment (NealSun96) [inline website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt:41]: *bug

File: website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+~~ under the License.
+
+~~ NOTE: For help with the syntax of this file, see:
+~~ http://maven.apache.org/guides/mini/guide-apt-format.html
+
+Release Notes for Apache Helix 0.9.8
+
+  The Apache Helix team would like to announce the release of Apache Helix 0.9.8.
+
+  This is the twenty-third release under the Apache umbrella, and the sixteenth as a top-level project.
```
Comment (NealSun96) [inline website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt:29]: Nit: 23rd and 16th?

**Review by NealSun96 — CHANGES_REQUESTED** [review]

File: website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt (hunk 1 above)
Comment (jiajunwang) [reply website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt:41]: Good catch!

File: website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt (hunk 2 above)
Comment (jiajunwang) [reply website/0.9.8/src/site/apt/releasenotes/release-0.9.8.apt:29]: That's the convention...

**Review by dasahcc — APPROVED** [review]
For release note, we dont have to review. Just add it and commit it in web folder.

**Pull request merged**